polkadot/parachain/tests/basic_add.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Basic parachain that adds a number as part of its state.
+
+extern crate polkadot_parachain as parachain;
+extern crate tiny_keccak;
+
+use parachain::ValidationParams;
+use parachain::codec::{Slicable, Input};
+
+// Head data for this parachain.
+#[derive(Default, Clone)]
+struct HeadData {
+	// Block number
+	number: u64,
+	// parent block keccak256
+	parent_hash: [u8; 32],
+	// hash of post-execution state.
+	post_state: [u8; 32],
+}
+
+impl Slicable for HeadData {
+	fn encode(&self) -> Vec<u8> {
```

Comment:
That's potentially deriveable, right? Should we log an issue to work on proc macro for this?

Reply:
yeah, we would want to split the trait into two as well.

Reply:
https://github.com/paritytech/polkadot/issues/146